modules/plugin/geopkg/src/test/java/org/geotools/geopkg/GeoPkgDatetimeTest.java

```diff
@@ -164,6 +164,22 @@ public void testGroupBy() throws IOException {
         assertEquals(java.sql.Date.valueOf("2020-02-20"), results.get(singletonList("2")));
     }
 
+    @Test
+    public void testUniqueTimestamp() throws IOException {
+        FilterFactory ff = CommonFactoryFinder.getFilterFactory(null);
+        UniqueVisitor visitor =
+                new UniqueVisitor(
+                        "datetime", gpkg.getFeatureSource(gpkg.getTypeNames()[0]).getSchema());
+
+        SimpleFeatureSource fs = gpkg.getFeatureSource(gpkg.getTypeNames()[0]);
+        SimpleFeatureCollection features = fs.getFeatures();
+        features.accepts(visitor, NULL_LISTENER);
+        List results = visitor.getResult().toList();
+        for (Object result : results) {
+            assertThat(result, CoreMatchers.instanceOf(Timestamp.class));
```

Comment:
Wanted to check the actual values, but it gets a bit complicated (for the short time I had):
* Order is not guaranteed
* Timezones need to be take into account for an equality test to work